In [2]:
import pandas as pd 
import numpy as np 

In [14]:
df = pd.read_csv(r'D:\Internship material\Natwest data science project\cubic_zirconia_original.csv')

In [34]:
df.dtypes

No.          int64
carat      float64
cut         object
color       object
clarity     object
depth      float64
table      float64
x          float64
y          float64
z          float64
price        int64
dtype: object

In [15]:
df.head() #to display first five elements of the dataframe

No.  carat        cut color clarity  depth  table     x     y     z  price
0    1   0.30      Ideal     E     SI1   62.1   58.0  4.27  4.29  2.66    499
1    2   0.33    Premium     G      IF   60.8   58.0  4.42  4.46  2.70    984
2    3   0.90  Very Good     E    VVS2   62.2   60.0  6.04  6.12  3.78   6289
3    4   0.42      Ideal     F     VS1   61.6   56.0  4.82  4.80  2.96   1082
4    5   0.31      Ideal     F    VVS1   60.4   59.0  4.35  4.43  2.65    779

In [16]:
df.tail() #to display last 5 elements of the dataframe

No.  carat        cut color clarity  depth  table     x     y     z  \
26962  26963   1.11    Premium     G     SI1   62.3   58.0  6.61  6.52  4.09   
26963  26964   0.33      Ideal     H      IF   61.9   55.0  4.44  4.42  2.74   
26964  26965   0.51    Premium     E     VS2   61.7   58.0  5.12  5.15  3.17   
26965  26966   0.27  Very Good     F    VVS2   61.8   56.0  4.19  4.20  2.60   
26966  26967   1.25    Premium     J     SI1   62.0   58.0  6.90  6.88  4.27   

       price  
26962   5408  
26963   1114  
26964   1656  
26965    682  
26966   5166

In [18]:
df.shape #to find rows and columns of the dataframe

(26967, 11)

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26967 entries, 0 to 26966
Data columns (total 11 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   No.      26967 non-null  int64  
 1   carat    26967 non-null  float64
 2   cut      26967 non-null  object 
 3   color    26967 non-null  object 
 4   clarity  26967 non-null  object 
 5   depth    26270 non-null  float64
 6   table    26967 non-null  float64
 7   x        26967 non-null  float64
 8   y        26967 non-null  float64
 9   z        26967 non-null  float64
 10  price    26967 non-null  int64  
dtypes: float64(6), int64(2), object(3)
memory usage: 2.3+ MB


In [25]:
df.isna().sum()

No.          0
carat        0
cut          0
color        0
clarity      0
depth      697
table        0
x            0
y            0
z            0
price        0
dtype: int64

In [21]:
df.describe()

No.         carat         depth         table             x  \
count  26967.000000  26967.000000  26270.000000  26967.000000  26967.000000   
mean   13484.000000      0.798375     61.745147     57.456080      5.729854   
std     7784.846691      0.477745      1.412860      2.232068      1.128516   
min        1.000000      0.200000     50.800000     49.000000      0.000000   
25%     6742.500000      0.400000     61.000000     56.000000      4.710000   
50%    13484.000000      0.700000     61.800000     57.000000      5.690000   
75%    20225.500000      1.050000     62.500000     59.000000      6.550000   
max    26967.000000      4.500000     73.600000     79.000000     10.230000   

                  y             z         price  
count  26967.000000  26967.000000  26967.000000  
mean       5.733569      3.538057   3939.518115  
std        1.166058      0.720624   4024.864666  
min        0.000000      0.000000    326.000000  
25%        4.710000      2.900000    945.000000  
50%        5.710000      3.520000   2375.000000  
75%        6.540000      4.040000   5360.000000  
max       58.900000     31.800000  18818.000000

In [39]:
df.describe(include="object") #includes object values

cut  color clarity
count   26967  26967   26967
unique      5      7       8
top     Ideal      G     SI1
freq    10816   5661    6571

EDA :<br>
1.The Data has 26,967 rows and 11 columns <br> 
2.Price is dependent variable rest all are independent variables and first column is just identifier or Sr. no. Three columns are object type cut , colour and clarity.<br>
3.There is 697 missing values in depth column.<br>
4.Insights from the df.describe of each column:<br>
    a. Carat:- 75% of diamonds have less than or equal to 1.05 carats(for further analysis plot should be made)<br>
    b. Depth:- Its values lies between 50.8% to 70.3% and 75% of all the depth are less than 62.50% depth.Since these are percentages mean does not have any signficance.<br>
    c.Table:- Its values lies between 49% to 79%. This is also expressed in percentage terms.75% of values are less than 59% table value.These has null values<br>
    d.In x y z - all the columns with zero values would be removed ,as there cannot be 2D zircona ,also there are going to be outlier , so we have to remove them.<br>
    e.Top values in each cut , colour and clarity are mentioned.<br>